File: Miet/divC/Training-1/A.py
# -*- coding: utf-8 -*-
# @Author: kostya
# @Date:   2021-11-13 17:21:28
# @Last Modified by:   kostya
# @Last Modified time: 2021-11-14 00:44:00
 
def wrap(s):
	if s == '':
		return []
	ans = [[s[0], 1]]
	for i in range(1, len(s)):
		if s[i] == ans[-1][0]:
			ans[-1][1] += 1
		else:
			ans.append([s[i], 1])
	return ans
 
for _ in range(int(input())):
	s = input()
	if not ('1' in s and '2' in s and '3' in s):
		print(0)
		continue
	ans = len(s)
	wr = wrap(s)
	for i in range(1, len(wr) - 1):
		if ''.join(sorted(wr[i - 1][0] + wr[i][0] + wr[i + 1][0])) == '123':
			ans = min(ans, wr[i][1] + 2)
	print(ans)

File: Miet/divC/Training-1/A.cpp
/*
* @Author: kostya
* @Date:   2021-11-14 00:04:34
* @Last Modified by:   kostya
* @Last Modified time: 2021-11-14 15:48:46
*/

#include <iostream>
#include <vector>
#include <string>
#include <set>

using namespace::std;

bool in(const string &st, char target)
{
	for (char c : st)
		if (c == target)
			return true;
	return false;
}

void wrap(const string &s, vector<pair<char, int>> &wr)
{
	if (s.empty())
		return;
	wr.push_back({s[0], 1});
	for (int i=1; i < s.size(); ++i)
	{
		if (s[i] == wr.back().first)
			++wr.back().second;
		else
			wr.push_back({s[i], 1});
	}
}

bool unique(char a, char b, char c)
{
	return a != b and a != c and b != c;
}

int main()
{
	int n;
	cin >> n;
	while (n--)
	{
		string s;
		cin >> s;
		if (not (in(s, '1') and in(s, '2') and in(s, '3')))
		{
			cout << 0 << endl;
			continue;
		}
		int ans = s.size();
		vector<pair<char, int>> wr;
		wrap(s, wr);
		for (int i=1; i < wr.size() - 1; ++i)
		{
			if (unique(wr[i - 1].first, wr[i].first, wr[i + 1].first))
				ans = min(ans, wr[i].second + 2);
		}
		cout << ans << endl;
	}
}
